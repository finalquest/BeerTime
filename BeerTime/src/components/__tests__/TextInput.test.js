import React from 'react';
import { shallow } from 'enzyme';
import Input from '../TextInput';


describe('TextInput test suite', () => {
  it('renders as expected with standar props', () => {
    const props = {
      label: 'label',
      labelStyle: {
        fontSize: 12,
      },
      style: {
        height: 30,
        borderBottomWidth: 1,
        borderColor: 'lightgray',
      },
      onChange: () => {},
      error: '',
    };
    const wrapper = shallow(<Input {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call onChange', () => {
    const onPress = jest.fn();
    const props = {
      label: 'label',
      labelStyle: {
        fontSize: 12,
      },
      style: {
        height: 30,
        borderBottomWidth: 1,
        borderColor: 'lightgray',
      },
      onChange: onPress,
    };
    const wrapper = shallow(<Input {...props} />);
    wrapper.find('TextInput').prop('onChangeText')();
    expect(onPress).toBeCalled();
  });
});
