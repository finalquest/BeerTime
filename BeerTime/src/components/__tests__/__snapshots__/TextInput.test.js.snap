// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput test suite renders as expected with standar props 1`] = `
<View
  style={
    Object {
      "alignSelf": "stretch",
    }
  }
>
  <Label
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 12,
      }
    }
    uppercase={false}
  >
    label
  </Label>
  <TextInput
    allowFontScaling={true}
    autoCorrect={false}
    keyboardType="default"
    onChangeText={[Function]}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "borderBottomWidth": 1,
          "borderColor": "lightgray",
          "height": 30,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
</View>
`;
